### ISSUE TITLE
Media Export 

### ISSUE THREAD

**Issue by WorldImpex** [ChildMindInstitute/mindlogger-backend#991]
Goal: 

Secure Storage, Transit and Admin Panel Export

**Comment by WorldImpex** [comment]
Blocked by 964

**Comment by natalia-muzyka** [comment]
How the geolocation should be exported?
In the CSV I see only the label "Geolocation" but I don't see any additional info about the location
https://www.screencast.com/t/Zh5L95NQ
[https://app.zenhub.com/files/103405378/ae05871e-1f11-42e3-ab4c-209d2843c309/download](https://app.zenhub.com/files/103405378/ae05871e-1f11-42e3-ab4c-209d2843c309/download)
@WorldImpex @devbtech

**Comment by WorldImpex** [comment]
1. @devbtech Please respond 
2. Yes
3. We do not have requirements for that export yet. 
4. We should solve that

**Comment by WorldImpex** [comment]
1. @natalia-muzyka It looks like geolocation works from @devbtech's end. Please confirm

https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-admin/1067

We still need to solve 4 though

**Comment by natalia-muzyka** [comment]
I've just tested media export and see that it doesn't work except for the SVG file.
See "null" responses in the exported CSV, also noticed the same "null" response for the usual radio button/checkbox/slider while was testing subscales + dataviz:
https://www.screencast.com/t/hnGTjhlEuBBr

Environment:
https://admin-staging.mindlogger.org/
ML v0.16.28 (staging)
Samsung Galaxy S7 Edge SM-G935FD // Android 8.0
iPhone XR // iOS 14.3

account: geria.test@yahoo.com//3FuuFz60
applet: applet list test
activity: media test
Applet password	Qwe123!!!

account: snakessqa@gmail.com // 123456
applet: media 
activity: media
Applet password	Qwe123!!!

Reproduces for newly created activity and for previously created,
for both platforms and both accounts.

For the "media" applet there are some files in the exported zip file but these are old files.
https://www.screencast.com/t/V9NcLUlHiLdc

All the latest exported data here:
https://drive.google.com/drive/folders/1SqMClwxuKLzNIrQZ4X-3nsOBQKHyCYMR?usp=sharing

**Comment by natalia-muzyka** [comment]
I'm still not able to get the files except the drawing (svg).
Geolocation I see now.

The "null" value for responses isn't fixed for:
- drawing (there is still "null" value for response and null activity start/end time)
- audio image + record has no value response at all

https://www.screencast.com/t/p12tNY2H

Reproduces for newly created activity and for previously created,
for both platforms and both accounts.

Environment:
https://admin-staging.mindlogger.org/
ML v0.16.28 (staging)
Samsung Galaxy S7 Edge SM-G935FD // Android 8.0
iPhone XR // iOS 14.3

account: geria.test@yahoo.com//3FuuFz60
applet: applet list test
activity: media test
Applet password Qwe123!!!

account: snakessqa@gmail.com // 123456
applet: media, activity: media
applet: new media, activity: new media
Applet password Qwe123!!!

The latest exported data for the newly created applet here :
https://drive.google.com/drive/folders/1qO-QFu2XvOYEooH7Wldgri02wJhVk8n-?usp=sharing

**Comment by WorldImpex** [comment]
@devbtech Export still does not work

**Comment by natalia-muzyka** [comment]
reproduced for a newly created account and applet:
account:
janedoeregress@protonmail.com	// %pVgPJ*O
applet: media new
applet password: Qwe123!!!

The latest exported data for the newly created applet here :
https://drive.google.com/drive/folders/1pvPRf4jSOwVBcD58SFHf6TYyN3Y-8qRr?usp=sharing

**Comment by natalia-muzyka** [comment]
I was able to get zip archive only once, now I'm getting only a csv file.
Clearing cache and hard reload don't help.
https://www.screencast.com/t/UwBcGPCUp

Accounts:
owner: johndoeregress@protonmail.com	// 12345678
user: janedoeregress@protonmail.com	// %pVgPJ*O
Applets:
media new
media test
Applet password Qwe123!!!

Win 10 // Chrome 91
ML v0.16.30 (staging)
Samsung Galaxy S7 Edge SM-G935FD // Android 8.0
iPhone XR // iOS 14.3

**Comment by natalia-muzyka** [comment]
I can export now all the media: photo, video, audio, drawing, location.

As previously:
On Android and iOS the uploading audio/video is implemented with a difference: iOS automatically compresses audio and video files, Android doesn't. Because of this in case of uploading a large file(10 Mb+), Android uploads the original data and it takes more time. If the admin wants to export a user's data there could be only CSV file downloaded (without zip) or the large file will be skipped.

**Comment by natalia-muzyka** [comment]
@devbtech 
The media export doesn't work again (for an existing applet/activity):
after completing the assessment with a photo, video, audio, drawing, geolocation response is displayed in the CSV only for the geolocation,
Downloaded ZIP archive contains only the previously added files.
Can't check with a newly created applet because of the [#1112](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-backend/1112)

**Video:** https://www.screencast.com/t/hZ4YcmoEPmj

**Environment:**
https://admin-staging.mindlogger.org/
Account:
janedoeregress@protonmail.com // %pVgPJ*O
johndoeregress@protonmail.com	// 12345678
Applets:
media test / media activity
Applet password Qwe123!!!

**Comment by natalia-muzyka** [comment]
Verified as fixed

Video: https://www.screencast.com/t/znsp9rWBY

Environment:
https://admin-staging.mindlogger.org/
ML v0.17.5 (staging)
Samsung Galaxy S7 Edge SM-G935FD // Android 8.0
iPhone 6 Plus  // iOS 12.3.1

Account:
janedoeregress@protonmail.com // %pVgPJ*O
johndoeregress@protonmail.com // 12345678
Applets:
media test / media activity
Applet password Qwe123!!!

**Comment by WorldImpex** [comment]
Removed the iOS/ Android bug into a separate ticket - [1186](https://app.zenhub.com/workspaces/mindlogger-webapp-5e11094d0c26311588da9626/issues/childmindinstitute/mindlogger-admin/1186)
